configure.ac

```diff
@@ -61,6 +61,7 @@ AC_ARG_ENABLE([debug],
                              [enable debugging output])],[],[enable_debug=yes])
 AM_CONDITIONAL([ENABLE_DEBUG], test x"$enable_debug" = x"yes")
 if test x"$enable_debug" = x"yes"; then
+   CFLAGS="${CFLAGS} -O0 -g3 -fsanitize=address -fno-omit-frame-pointer"
```

Comment:
Running `make check` with address sanitizer on makes running tests on my machine _a lot_ slower. We should put turning ASAN on behind a different flag, say `--enable-asan`, or just pass `CFLAGS` and `LDFLAGS` manually from `test.sh` (which I would call `setup.sh` or some such, since it's not really a test, but setup for tests)